### DISCUSSION TITLE
Resources Page is 404ing

### DISCUSSION THREAD

**Discussion by K0nja** [nmelhado/league-page#284] [Help!] [answered]
I noticed the resources page is 404ing. I have the newest stuff from the master repo and main fork all that. 

Here is my league's page: https://leagueofhaiwans.com/

any clue what's up?

**Comment by Drewmaloney** [comment]
Hello, yeah, I'm having this issue as well all of a sudden. Was working ok then started 404 out of nowhere. 

I removed the leading ... from news.js line 27 

`	const articles = [reddit, serverData].sort((a, b) => (a.ts < b.ts) ? 1 : -1);
	news.update(() => articles);
`

in front of reddit and serverData and that stopped giving me a 404 and the resources are back. But the news and updates are still broken. 
![image](https://github.com/nmelhado/league-page/assets/158063108/56d3678d-f353-4286-9b0d-e6beeb7a0903)


Doing some debugging it says the issue lies here
`	const newsSources = [
		smartFetch(SERVER_API, {compress: true}), 
	];
`

So, I'm guessing it's an issue in routes/api/fetch_serverside_news/server.js but I can't pinpoint exactly what. Perhaps some endpoint changed but I can't quite get there. If you have discovered anything please share! 

Thanks, 
Drew

**Reply by nmelhado** [reply]
Going to look into this and a few other issues tonight. Should have some updates out shortly.

**Reply by Drewmaloney** [reply]
So, I think it was the result of a broken link that was throwing everything off. FTN_NEWS. So, I just commented that out.
```
export async function GET() {
	const articles = [
        getXMLArticles(FF_BALLERS, processFF),
/*         getJSONArticles(FTN_NEWS, processFTN), */
	];
	if(dynasty) {
		articles.push(getXMLArticles(DYNASTY_LEAGUE, processDynastyLeague));
		articles.push(getXMLArticles(DYNASTY_NERDS, processDynastyNerds));
	} 
```


and in news.js flattened articles. Also, kept the leading ... off reddit + serverData. 
``` 	
const articles = [reddit, serverData].flat().sort((a, b) => (a.ts < b.ts) ? 1 : -1);
	news.update(() => articles); 
``` 

I obvi don't know if this was the correct or right fix, but I am now getting intended behavior. You can check:
https://www.grindersgridirongang.com/

HOWEVER, if you go directly to https://www.grindersgridirongang.com/resources you will get a 404. You have to go to HOME first then resources. and if you reload on resources, you'll get 404'd again. 
![image](https://github.com/nmelhado/league-page/assets/158063108/a58831e7-64ec-46ad-9b32-fb3133e7c6e0)

![image](https://github.com/nmelhado/league-page/assets/158063108/088b57a1-dd09-4c83-8f21-bce90cf2ac7c)


I think it's the doctype which may have been the underlying issue all along? 

Let me know your thoughts, I'm still very new to this.

**Reply by nmelhado** [reply, marked as answer]
This is fixed in the latest update. There were two issues, one was the FTN one you found and the other was an SSR fetch issue. Everything should be working now though!
